scylla-cdc/src/consumer.rs

```diff
@@ -0,0 +1,474 @@
+use num_enum::TryFromPrimitive;
+use scylla::frame::response::result::{ColumnSpec, CqlValue, Row};
+use std::collections::HashMap;
+
+pub trait Consumer {
+    fn consume_cdc(&mut self, data: CDCRow);
+}
+
+pub trait ConsumerFactory {
+    fn new_consumer() -> Box<dyn Consumer>;
+}
+
+#[derive(Debug, Eq, PartialEq, TryFromPrimitive)]
+#[repr(i8)]
+pub enum OperationType {
+    PreImage,
+    RowUpdate,
+    RowInsert,
+    RowDelete,
+    PartitionDelete,
+    RowRangeDelInclLeft,
+    RowRangeDelExclLeft,
+    RowRangeDelInclRight,
+    RowRangeDelExclRight,
+    PostImage,
+}
+
+pub struct CDCRowSchema {
+    pub(crate) stream_id: usize,
```

Comment:
These values are indices of these columns in the CDC log table schema, or something else? A comment should mention this. (I didn't understand what these `usize`s mean until I read the implementation of `new`)